packages/repeater/src/lib/RepeaterFactory.spec.ts

```diff
@@ -0,0 +1,62 @@
+import 'reflect-metadata';
+import { RepeaterFactory } from './RepeaterFactory';
+import { Repeater } from './Repeater';
+import { RepeatersManager } from '../api';
+import { EventBusFactory } from '../bus';
+import { Configuration, EventBus } from '@secbox/core';
+import { anything, instance, mock, reset, when } from 'ts-mockito';
+import { DependencyContainer } from 'tsyringe';
+
+describe('RepeaterFactory', () => {
+  const repeaterId = 'fooId';
+
+  const MockedContainer = mock<DependencyContainer>();
+  const MockedConfiguration = mock<Configuration>();
+  const MockedRepeaterManager = mock<RepeatersManager>();
+  const MockedEventBusFactory = mock<EventBusFactory>();
+
+  const mockedConfiguration = instance(MockedConfiguration);
+
+  beforeEach(() => {
+    when(MockedContainer.resolve<EventBusFactory>(EventBusFactory)).thenReturn(
+      instance(MockedEventBusFactory)
+    );
+    when(
+      MockedContainer.resolve<RepeatersManager>(RepeatersManager)
+    ).thenReturn(instance(MockedRepeaterManager));
+
+    when(MockedEventBusFactory.create(anything())).thenResolve(
+      {} as unknown as EventBus
```

Comment:
Return a mocked instance, to make sure that `init()` is called when it is present

Reply:
stuck with issue similar to ts-mockito 206:

<details>
<summary>Fails</summary>

```ts
import { EventBusFactory } from '../bus';
import { anyString, instance, mock, when } from 'ts-mockito';
import { EventBus } from '@secbox/core';

describe('MockedEventBus', () => {
  const mockedEventBus = mock<EventBus>();
  const mockedEventBusFactory = mock<EventBusFactory>();

  const eventBusFactory = instance(mockedEventBusFactory);
  const eventBus = instance(mockedEventBus);

  beforeEach(() => {
    when(mockedEventBusFactory.create(anyString())).thenResolve(eventBus);
  });

  it.only('create bus', async () => {
    await eventBusFactory.create('foo');

    expect(true).toBeTruthy();
  });
});
```
</details>

that's why implementing desired check with jest...

Reply:
Yeah, that is well-known issue (see https://github.com/NagRock/ts-mockito/issues/191). You just need to use `resolvableInstance()` instead of `instance()`:

```
const resolvableInstance = <T extends object>(x: T): T => new Proxy<T>(instance(x), {
  get(target, prop, receiver) {
    if (["Symbol(Symbol.toPrimitive)", "then", "catch"].includes(prop.toString())) {
      return undefined;
    }

    return Reflect.get(target, prop, receiver);
  }
});

beforeEach(() => {
  when(mockedEventBusFactory.create(anyString())).thenResolve(resolvableInstance(mockedEventBus));
});
```

Reply:
reworked